lib/lightning/version_control/github_client.ex

```diff
@@ -0,0 +1,85 @@
+defmodule Lightning.VersionControl.GithubClient do
+  @moduledoc """
+  Tesla github http client we use this to make any network requests 
+  to github from Lightning
+  """
+  use Tesla
+  alias Lightning.VersionControl.GithubToken
+
+  plug(Tesla.Middleware.BaseUrl, "https://api.github.com")
+  plug(Tesla.Middleware.JSON)
```

Comment:
Are these actually used, my understanding around `use Tesla` is that you use the `get`, `post`/`request` functions that are injected into the module. By calling `Tesla.client` is that not just creating a new client/going around the functions that were injected?